channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -357,6 +361,15 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
             _self._xhrReadyStateChange = (xhr: XMLHttpRequest, payload: string[], countOfItemsInPayload: number) => {
                 if (xhr.readyState === 4) {
                     let response: IBackendResponse = null;
+
+                    // check if the xhr's responseURL is same as endpoint url
+                    // TODO after 10 redirects force send telemetry with'redirect=false' as query parameter. 
+                    if(xhr.responseURL !== _self._senderConfig.endpointUrl()) {
```

Comment:
Probably should check that
- the responseURL is not empty and look valid (some XHR polyfill's might not set this etc)

Additional considerations.
- Do we want to retry the original URL if the redirected one also continues to fail
- Do we only want to update on a successful send (200-299 response status code)

Reply:
I think updating the url only after we get a successful xhr send is a better approach.